[Miro] connector improvements (#22)
cohere-ai/quick-start-connectors: miro/README.md

To configure this connector you will have to either create a new Miro app, see [here](https://developers.miro.com/docs/rest-api-build-your-first-hello-world-app#step-1-create-your-app-in-miro), or configuring an existing app.

> Your app will require at the least the `board:read` permission to allow searching your boards, once that is done you can save and install the app,
> a message should notify you that the app was successfully installed, and give you an access token. Use this token for the `MIRO_ACCESS_TOKEN` environment variable.

Reviewer: ```suggestion
Your app will require at the least the `board:read` permission to allow searching your boards, once that is done you can save and install the app.
A message should notify you that the app was successfully installed, and give you an access token. Use this token for the `MIRO_ACCESS_TOKEN` environment variable.
```

Author: Thank you

The `MIRO_ACCESS_TOKEN` environment variable must be set if you didn't select the "Expire user authorization token" checkbox during app creation.
If you selected that checkbox, you will need to use the Oauth flow.

> ### Oauth flow

Reviewer: ```suggestion
### OAuth flow
```

Author: Thanks

> The OAuth flow should occur outside of the Connector and Cohere's API will forward the user's access token

Reviewer: ```suggestion
The OAuth flow should occur outside of the Connector by calling Cohere's API. Once authorized, Cohere's API will forward the user's access token
```

Author: Thanks

to this connector through the `Authorization` header.

To configure OAuth, follow the same steps to create a Miro App and select the "Expire user authorization token" checkbox. 
You will also need to register a redirect URI on that app to `https://api.cohere.com/v1/connectors/oauth/token`.

You can then register the connector with Cohere's API using the following configuration:
Note: Your app `CLIENT-ID` and app `CLIENT-SECRET` values correspond to `client_id` and `client_secret` respectively.

```bash
curl  -X POST \
  'https://api.cohere.ai/v1/connectors' \
  --header 'Accept: */*' \
  --header 'Authorization: Bearer {COHERE-API-KEY}' \
  --header 'Content-Type: application/json' \
  --data-raw '{
  "name": "MS Teams with OAuth",
  "url": "{YOUR_CONNECTOR-URL}",
  "oauth": {
    "client_id": "{Your App CLIENT-ID}",
    "client_secret": "{Your App CLIENT-SECRET}",
    "authorize_url": "https://miro.com/oauth/authorize",
    "token_url": "https://api.miro.com/v1/oauth/token",
  }
}'
```
No more configuration is needed here and after successfully registration, 

> Cohere will take care of OAuth steps including passing in the correct headers to your connector.

Reviewer: We might want to mention that users will have to complete the authorization flow here (if they're not using coral/playground) by navigating to the url returned at `/connectors/{id}/oauth/authorize`

Author: Thanks
